test/framework/autoscaler_helpers.go

```diff
@@ -0,0 +1,438 @@
+		return nil, err
+	}
+
+	return out, nil
+}
+
+type DisableAutoscalerForMachineDeploymentTopologyAndWaitInput struct {
+	ClusterProxy                  ClusterProxy
+	Cluster                       *clusterv1.Cluster
+	WaitForAnnotationsToBeDropped []interface{}
+}
+
+// DisableAutoscalerForMachineDeploymentTopologyAndWait drop the autoscaler annotations from the MachineDeploymentTopology
+// and waits till the annotations are dropped from the underlying MachineDeployment. It also verifies that the replicas
+// fields of the MachineDeployments are not affected after the annotations are dropped.
+func DisableAutoscalerForMachineDeploymentTopologyAndWait(ctx context.Context, input DisableAutoscalerForMachineDeploymentTopologyAndWaitInput) {
+	Expect(ctx).NotTo(BeNil(), "ctx is required for DisableAutoscalerForMachineDeploymentTopologyAndWait")
+	Expect(input.ClusterProxy).ToNot(BeNil(), "Invalid argument. input.ClusterProxy can't be nil when calling DisableAutoscalerForMachineDeploymentTopologyAndWait")
+
+	mgmtClient := input.ClusterProxy.GetClient()
+
+	// Get the current replicas of the MachineDeployments.
+	replicas := map[string]*int32{}
+	mdList := GetMachineDeploymentsByCluster(ctx, GetMachineDeploymentsByClusterInput{
+		Lister:      mgmtClient,
+		ClusterName: input.Cluster.Name,
+		Namespace:   input.Cluster.Namespace,
+	})
+	for _, md := range mdList {
+		replicas[md.Name] = md.Spec.Replicas
+	}
+
+	log.Logf("Dropping the %s and %s annotations from the MachineDeployments in ClusterTopology", clusterv1.AutoscalerMinSizeAnnotation, clusterv1.AutoscalerMaxSizeAnnotation)
+	patchHelper, err := patch.NewHelper(input.Cluster, mgmtClient)
+	Expect(err).ToNot(HaveOccurred())
+	for i := range input.Cluster.Spec.Topology.Workers.MachineDeployments {
+		md := input.Cluster.Spec.Topology.Workers.MachineDeployments[i]
+		delete(md.Metadata.Annotations, clusterv1.AutoscalerMinSizeAnnotation)
+		delete(md.Metadata.Annotations, clusterv1.AutoscalerMaxSizeAnnotation)
+		input.Cluster.Spec.Topology.Workers.MachineDeployments[i] = md
+	}
+	Eventually(func(g Gomega) {
+		g.Expect(patchHelper.Patch(ctx, input.Cluster)).Should(Succeed())
+	}, retryableOperationTimeout, retryableOperationInterval).Should(Succeed(), "Failed to patch Cluster topology to drop autoscaler annotations")
+
+	log.Logf("Wait for the annotations to be dropped from the MachineDeployments")
+	Eventually(func(g Gomega) {
+		mdList := GetMachineDeploymentsByCluster(ctx, GetMachineDeploymentsByClusterInput{
+			Lister:      mgmtClient,
+			ClusterName: input.Cluster.Name,
+			Namespace:   input.Cluster.Namespace,
+		})
+		for i := range mdList {
+			md := mdList[i]
+			g.Expect(md.Annotations).ToNot(HaveKey(clusterv1.AutoscalerMinSizeAnnotation))
+			g.Expect(md.Annotations).ToNot(HaveKey(clusterv1.AutoscalerMaxSizeAnnotation))
+			// Verify that disabling auto scaler does not change the current MachineDeployment replicas.
+			g.Expect(md.Spec.Replicas).To(Equal(replicas[md.Name]))
```

Comment:
just a note here, we've found in our openshift tests that it can difficult to figure out why something is failing, we might consider surfacing these values to the outer `Eventually` call as a future improvement.

i'm certainly not a ginkgo/gomega expert though, so if these errors will print in the final report then no worries.

Reply:
They should surface automatically

Reply:
we can also add an error message after expect, so it is easier to put the error into context when triaging (we are doing this incrementally across the entire test framework)

Reply:
I should surface automatically (I believe because the function passed into `Eventually` has `g` as an argument.).

As Fabrizio suggested, I will also add some descriptions to the assertion statements. Should make the output richer and easier to debug incase of a failure.